token/program-2022/src/processor.rs

```diff
@@ -157,6 +157,13 @@ impl Processor {
         {
             msg!("Warning: Mint has a permanent delegate, so tokens in this account may be seized at any time");
         }
+
+        if mint.get_extension::<NonTransferable>().is_ok()
+            && account.get_extension::<ImmutableOwner>().is_err()
+        {
+            return Err(TokenError::NonTransferableNeedsImmutableOwnership.into());
+        }
```

Comment:
Yes, I agree. 
But if we remove the check, we will return the InvalidAccountData error if the account doesn't provide enough space to store the ImmutableOwnership extension while initializing a non-transferable account, 
For this scenario, I think the InvalidAccountData error is less explicit than the NonTransferableNeedsImmutableOwnership error. 
Is it necessary to give a more specific error?

Reply:
I would say that since we're changing the behavior to automatically initialize the immutable owner extension, the new error is actually less clear. The true error is now not having a big enough account. So I think that behavior is fine